mitodl/open-discussions
# Added support for tracking and displaying popular learning resources

#### Pre-Flight checklist

![Screenshot_2020-02-07 MIT Open Learning](https://user-images.githubusercontent.com/28598/74069142-66329080-49cb-11ea-974a-b72e72a4eaa8.png)
![Screen Shot 2020-02-07 at 17 00 36](https://user-images.githubusercontent.com/28598/74069129-629f0980-49cb-11ea-93dd-8af22471fdde.png)

#### What are the relevant tickets?

Closes #2582

#### What's this PR do?

- Adds data model and API for tracking user interactions over content objects
- Adds API to return a list of popular content
- Adds frontend code to record a view interaction on drawer open
- Adds frontend code to display the popular learning resources on the /learn page

#### How should this be manually tested?

- Go to /learn before you open the drawer for a resource, verify that with no interactions tracked the page still displays
- Click on a few of the resources in /learn and /learn/search and verify those fire view interaction requests to the backend
- Go back to /learn and verify that popular items are being returned and they reorder when popularity changes
